lib.sh

```diff
@@ -16,6 +16,49 @@ then
 	exec {__sleep_fd}<> <(:)
 fi
 
+typeof() {
+	# typeof -- returns the type of a variable
+
+	local type_sig
+	type_sig=$(declare -p "${1}" 2>/dev/null)
+	if [[ "${type_sig}" =~ "declare --" ]]
+	then
+		str_type "${1}"
+	elif [[ "${type_sig}" =~ "declare -a" ]]
+	then
+		printf "array"
+	elif [[ "${type_sig}" =~ "declare -A" ]]
+	then
+		printf "map"
+	else
+		printf "none"
+	fi
+}
+
+str_type() {
+	# str_type -- returns the type of a string
+
+	local -n str="${1}"
+
+	if [[ "${str}" =~ ^[0-9]+$ ]]
```

Comment:
Is it not possible in bash to make a case out of this? Or is the level of regular expression permitted in case statements not sufficient?